src/nbl/builtin/builtinDataGen.py

```diff
@@ -17,18 +17,15 @@
     cmakeSourceDir = sys.argv[2]
     resourcesFile  = sys.argv[3]
     resourcesNamespace = sys.argv[4]
+    correspondingHeaderFile = sys.argv[5]
 
     with open(resourcesFile, "r") as f:
         resourcePaths = f.read().rstrip().split(',')
 
     #opening a file
     outp = open(outputFilename,"w+")
-  
-
-    outp.write("#include \"nbl/core/string/StringLiteral.h\"\n")
-    outp.write("#include <cstdint>\n")
-    outp.write("#include <unordered_map>\n");
-    outp.write("#include <string>\n");
+    
+    outp.write("#include \"" + correspondingHeaderFile + "\"\n")
```

Comment:
I think this is what fucks up the `extern` on the #488 PR, you can't have `extern template` in the translation unit that actually defines the template AFAIK

Reply:
yeah possibly, I will keep it as in #488 PR and have declaration of an explicit template specialization, this way it works